File: api/emojis.py
import requests 

BOTID = "1294045105594040431"
BOTTOKEN = "MTI5NDA0NTEwNTU5NDA0MDQzMQ.GpNU3V.eCXs7ReqVTL7CupNxvjpfj5W86OiAFODyRXDlk" 


def getEmojis(url=f"https://discord.com/api/v10/applications/{BOTID}/emojis") -> dict | None:

    try:
        headers = {
            "Authorization": f"Bot {BOTTOKEN}",
            "Content-Type": "application/json"
        }
            
        response = requests.get(url, headers=headers)
        data = response.json()

        if data is None:
            return None

        items = data.get("items", None)       
        emotes = {emoji["name"]: emoji["id"] for emoji in items}
        
        return emotes
        
    except Exception:
        return

File: src/discord/emojis.ts
import { CONFIG } from "@app/config";
import { getData } from "@lib";
import { API_URLS } from "@btd6";

interface EmojiData {
    id: string;
    name: string;
    user: Object;
    roles: [];
    required_colons: boolean;
    managed: boolean;
    animated: boolean;
    available: boolean;
}

interface EmojiResponse {
    items: EmojiData[];
}

interface FormattedEmojiData {
    id: string;
    name: string;
}

const EmojiCache = new Map<string, FormattedEmojiData>();

export async function loadEmojis(): Promise<void> {

  try {
    const data = await getData<EmojiResponse>(
      API_URLS.Emojis.replace("{}", CONFIG.BOT_ID),
        {
          Authorization: `Bot ${CONFIG.BOT_TOKEN}`,
          "Content-Type": "application/json"
        }
    )

    for (const emoji of data?.items) {
      EmojiCache.set(emoji.name, emoji)
    }
    console.log("Emojis loaded");
  } catch (error) {
    console.warn("Failed to load Emojis", error);
  }
}

export function getEmoji(name: string): FormattedEmojiData | undefined {

  const emoji = EmojiCache.get(name);

  if (!emoji) {
    console.warn(`Emoji ${name} wasn't found.`);
    return undefined;
  }
  return {
    id: emoji.id,
    name: emoji.name
  }
}